lib/new_relic/agent/http_clients/abstract.rb

```diff
@@ -37,6 +37,10 @@ def host
         def method
           raise NotImplementedError, MUST_IMPLEMENT_ERROR % [self.class, __method__]
         end
+
+        def headers
+          raise NotImplementedError, MUST_IMPLEMENT_ERROR % [self.class, __method__]
+        end
```

Comment:
I suppose we could technically DRY this up with something that reuses everything except the method name but that might be overkill...

```ruby
        def method
          not_implemented(__method__)
        end

        def headers
          not_implemented(__method__)
        end

        private

        def not_implemented(method_name)
          raise NotImplementedError, MUST_IMPLEMENT_ERROR % [self.class, method_name]
        end
```

Reply:
Good idea! I'll put that in a separate technical debt ticket/PR to avoid overloading the fix.

Reply:
Changed my mind! Fixed in 0b28bf78cf699ce57f346cf26fda78585d76aa96

Reply:
Using `define_method` is a slick approach. The new `not_implemented` method requires an argument though, and the callers don't appear to be providing one.

Reply:
Great catch! Fixed. I was also missing the URI method.

Reply:
edd33782b5a87d86a9ec671da3b3afd66b5f5f45

Reply:
Fix commit: 236788a0bef6043eb2c8caf7658f9576aa11ade4